[0.8.0] ClassNotFoundException: org.factcast.core.snap.Snapshot

We moved the class to another location which breaks client deserialization. 

Current idea: duplicate the class at the old location

```
Caused by: java.lang.ClassNotFoundException: org.factcast.core.snap.Snapshot
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:592)
	at org.springframework.boot.loader.net.protocol.jar.JarUrlClassLoader.loadClass(JarUrlClassLoader.java:103)
	at org.springframework.boot.loader.launch.LaunchedClassLoader.loadClass(LaunchedClassLoader.java:91)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at org.jboss.marshalling.AbstractClassResolver.loadClass(AbstractClassResolver.java:129)
	at org.jboss.marshalling.AbstractClassResolver.resolveClass(AbstractClassResolver.java:110)
	at org.jboss.marshalling.river.RiverUnmarshaller.doReadClassDescriptor(RiverUnmarshaller.java:1227)
	at org.jboss.marshalling.river.RiverUnmarshaller.doReadNewObject(RiverUnmarshaller.java:1560)
	at org.jboss.marshalling.river.RiverUnmarshaller.doReadObject(RiverUnmarshaller.java:441)
	at org.jboss.marshalling.river.RiverUnmarshaller.doReadObject(RiverUnmarshaller.java:373)
	at org.jboss.marshalling.AbstractObjectInput.readObject(AbstractObjectInput.java:41)
	at org.redisson.codec.MarshallingCodec.lambda$new$0(MarshallingCodec.java:147)
```

--- a/factcast-core/src/main/java/org/factcast/core/snap/Snapshot.java
+++ b/factcast-core/src/main/java/org/factcast/core/snap/Snapshot.java
@@ -1,0 +1,36 @@
+/*
+ * Copyright © 2017-2020 factcast.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.factcast.core.snap;
+
+import java.io.Serializable;
+import java.util.UUID;
+import lombok.NonNull;
+import lombok.Value;
+
+/**
+ * This is a copy of org.factcast.factus.snapshot.Snapshot to mitigate client side deserialization
+ * issues when updating to 0.8.0
+ */
+@Value
+public class Snapshot implements Serializable {
+  @NonNull SnapshotId id;
+
+  @NonNull UUID lastFact;
+
+  @NonNull byte[] bytes;
+
+  boolean compressed;
+}

--- a/factcast-core/src/main/java/org/factcast/core/snap/SnapshotId.java
+++ b/factcast-core/src/main/java/org/factcast/core/snap/SnapshotId.java
@@ -1,0 +1,34 @@
+/*
+ * Copyright © 2017-2020 factcast.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.factcast.core.snap;
+
+import java.io.Serializable;
+import java.util.UUID;
+import lombok.NonNull;
+import lombok.Value;
+
+/**
+ * This is a copy of org.factcast.factus.snapshot.SnapshotId to mitigate client side deserialization
+ * issues when updating to 0.8.0
+ */
+@Value(staticConstructor = "of")
+public class SnapshotId implements Serializable {
+  private static final long serialVersionUID = -3207528229703207635L;
+  @NonNull String key;
+
+  @NonNull UUID uuid; // semantically optional, might be constant if not needed
+  // (SnapshotProjections)
+}

